src/producer.ts

```diff
@@ -23,53 +22,55 @@ export class Producer {
-              const beatType = childElem.getElementsByTagName('beat-type').item(0)?.textContent ?? '';
-              messages.push({ type: 'timeSignature', top: beats, bottom: beatType });
-              break;
-            default:
-              console.log(`unprocessed node, got: ${children[i]}`);
-          }
+        const width = nodeElem.getAttribute('width') ?? '';
+        const staves = nodeElem.getElementsByTagName('staves').item(0)?.textContent ?? '1';
+        const sign: string[] = [];
+        for (let i = 0; i < parseInt(staves); i++) {
+          sign.push(nodeElem.getElementsByTagName('sign').item(i)?.textContent ?? '');
         }
+        const beats = nodeElem.getElementsByTagName('beats').item(0)?.textContent ?? '';
+        const beatType = nodeElem.getElementsByTagName('beat-type').item(0)?.textContent ?? '';
+        messages.push({ msgType: 'measureStart', width, staves, clef: sign, time: `${beats}/${beatType}` });
+        messages.push(...Array.from(node.childNodes).flatMap(this.getMessages));
+        messages.push({ msgType: 'measureEnd' });
+        break;
+      case 'attributes':
+        // Nothing to do processed in measure
         break;
       case 'note':
+        const rest = nodeElem.getElementsByTagName('rest').item(0) ? true : false;
         const step = nodeElem.getElementsByTagName('step').item(0)?.textContent ?? '';
         const octave = nodeElem.getElementsByTagName('octave').item(0)?.textContent ?? '';
         const stem = nodeElem.getElementsByTagName('stem').item(0)?.textContent ?? '';
         const accidental = nodeElem.getElementsByTagName('accidental').item(0)?.textContent ?? '';
-        const duration = nodeElem.getElementsByTagName('type').item(0)?.textContent ?? '';
+        const duration = nodeElem.getElementsByTagName('duration').item(0)?.textContent ?? '';
+        const type = nodeElem.getElementsByTagName('type').item(0)?.textContent ?? '';
+        const voice = nodeElem.getElementsByTagName('voice').item(0)?.textContent ?? '1';
+        const staff = nodeElem.getElementsByTagName('staff').item(0)?.textContent ?? '1';
         messages.push({
-          type: 'note',
-          stem: stem,
+          msgType: 'note',
+          rest,
+          stem,
           pitch: `${step}${octave}`,
-          duration: duration,
-          accidental: accidental,
+          duration,
+          type,
+          accidental,
+          voice,
+          staff,
         });
         const beam = '' + nodeElem.getElementsByTagName('beam').item(0)?.textContent;
-        if (beam === 'begin') messages.push({ type: 'beamStart' });
-        if (beam === 'end') messages.push({ type: 'beamEnd' });
+        if (beam === 'begin') messages.push({ msgType: 'beamStart' });
+        if (beam === 'end') messages.push({ msgType: 'beamEnd' });
         break;
       default:
-        console.log(`unprocessed node, got: ${node}`);
+        console.log(`unprocessed node, got:`, node);
```

Comment:
We don't have to do this now in this PR, but understanding what nodes we're skipping is crucial to understanding what we're not accounting for. We should eventually try to format this to have line numbers, maybe show a snippet of the XML, etc. to help us in the future.

Reply:
Or even better, surface this information so that we can render it in the dev app. I can take this on.

Reply:
I leave that to you. I cooment out for now.